integrations/operator/embeddedtbot/bot.go

```diff
@@ -0,0 +1,138 @@
+package embeddedtbot
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/gravitational/trace"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/gravitational/teleport/api/client"
+	"github.com/gravitational/teleport/api/client/proto"
+	"github.com/gravitational/teleport/lib/tbot"
+	"github.com/gravitational/teleport/lib/tbot/config"
+)
+
+type EmbeddedBot struct {
+	cfg *config.BotConfig
+
+	errGroup *errgroup.Group
+
+	credential *config.UnstableClientCredentialOutput
+}
+
+func New(botConfig *BotConfig) (*EmbeddedBot, error) {
+	credential := &config.UnstableClientCredentialOutput{}
+
+	cfg := (*config.BotConfig)(botConfig)
+	cfg.Storage = &config.StorageConfig{Destination: &config.DestinationMemory{}}
+	cfg.Outputs = []config.Output{credential}
+
+	err := cfg.CheckAndSetDefaults()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	bot := &EmbeddedBot{
+		cfg:        cfg,
+		credential: credential,
+	}
+
+	return bot, nil
+
+}
+
+func (b *EmbeddedBot) Preflight(ctx context.Context) (*proto.PingResponse, error) {
```

Comment:
Missing godoc
Are we using this for anything else other than for checking server features?
If so, I think we could change the function name to something more obvious to that